app/src/main/java/com/calendar/Activity/CountDays.java

```diff
@@ -0,0 +1,153 @@
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Objects;
+
+public class CountDays extends AppCompatActivity {
+
+    private static final DatabaseReference databaseReference = FirebaseDatabase.getInstance()
+            .getReference("Calendar");
+
+    private Button dataRangeButton;
+    private TextView dateRangeText;
+    private TextView numberOfFreeDaysText;
+    private TextView numberOfWorkingDaysText;
+    private MaterialDatePicker<Pair<Long, Long>> materialDatePicker;
+    private int numberOfFreeDays = 0;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.count_days);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        setGuiComponents();
+        setMaterialDatePicker();
+    }
+
+    private void setGuiComponents() {
+        dataRangeButton = findViewById(R.id.dateRangeButton);
+        dateRangeText = findViewById(R.id.dateRangeText);
+        numberOfFreeDaysText = findViewById(R.id.numberOfFreeDays);
+        numberOfWorkingDaysText = findViewById(R.id.numberOfWorkingDays);
+    }
+
+    private void setMaterialDatePicker() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.DAY_OF_MONTH, 1);
+        long monthStart = calendar.getTimeInMillis();
+        calendar.set(Calendar.DAY_OF_MONTH, calendar.getActualMaximum(Calendar.DAY_OF_MONTH));
+        long monthEnd = calendar.getTimeInMillis();
+
+        materialDatePicker = MaterialDatePicker.Builder
+                .dateRangePicker()
+                .setSelection(new Pair<>(monthStart, monthEnd))
+                .build();
+        setMaterialDatePickerListener();
+        materialDatePicker.show(getSupportFragmentManager(), "tag_picker");
+    }
+
+    private void setMaterialDatePickerListener() {
+        materialDatePicker.addOnPositiveButtonClickListener(selection -> {
+            dateRangeText.setText(materialDatePicker.getHeaderText());
+            Date startDate = new Date(selection.first);
+            Date endDate = new Date(selection.second);
+            LocalDate startLocalDate = dateToLocalDate(startDate);
+            LocalDate endLocalDate = dateToLocalDate(endDate);
+            countDays(startLocalDate, endLocalDate);
+        });
+    }
+
+    public void selectDateRange(View view) {
+        materialDatePicker.show(getSupportFragmentManager(), "tag_picker");
+    }
+
+    private LocalDate dateToLocalDate(Date date) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        int year = calendar.get(Calendar.YEAR);
+        int month = calendar.get(Calendar.MONTH) + 1;
+        int day = calendar.get(Calendar.DAY_OF_MONTH);
+        return LocalDate.of(year, month, day);
+    }
+
+    public void countDays(LocalDate startDate, LocalDate endDate) {
```

Comment:
This method makes three things: count free days, count all days and set text on view...method name also doesn't say anything about setting views. I don't like it.

Reply:
I simplified this and logically split it into 2 functions.